app/screens/thread/thread_post_list/thread_post_list.tsx

```diff
@@ -34,18 +38,38 @@ const styles = StyleSheet.create({
 
 const ThreadPostList = ({
     channelLastViewedAt, isCRTEnabled,
-    nativeID, posts, rootPost, teamId, thread,
+    nativeID, posts, rootPost, teamId, thread, version,
 }: Props) => {
     const isTablet = useIsTablet();
     const serverUrl = useServerUrl();
 
+    const canLoadPosts = useRef(true);
+    const fetchingPosts = useRef(false);
+    const onEndReached = useCallback(debounce(async () => {
+        if (isMinimumServerVersion(version, 6, 6) && !fetchingPosts.current && canLoadPosts.current && posts.length) {
+            fetchingPosts.current = true;
+            const options: FetchPaginatedThreadOptions = {};
+            const lastPost = posts[posts.length - 1];
+            if (lastPost) {
+                options.fromPost = lastPost.id;
+                options.fromCreateAt = lastPost.createAt;
+            }
+            const result = await fetchPostThread(serverUrl, rootPost.id, options);
+            fetchingPosts.current = false;
+            canLoadPosts.current = false;
+            if (!('error' in result)) {
+                canLoadPosts.current = (result.posts?.length ?? 1) > 0;
```

Comment:
The logic here is not clear. From my understanding, this could be rewritten to something like:
```
if (!result.post) {
    canLoadPost.current = true;
} else {
    canLoadPost.current = Boolean(result.posts.length);
}
```
Am I right? Is this intended?
I am not saying we should use my version, but the current version seems a bit confusing.

Reply:
* if you don't have any error in the result -> you can load more posts 
* if the length of result.posts is zero - we have reached the end of lists -> we cannot load more posts
* if we don't have error and the length of results.posts is not zero, then we can load more posts

Is my understanding correct?  I agree that this section can be refactored.

Reply:
@avinashlng1080 you are right. It's the same code as channels as it was handling the use cases.

Reply:
@anurag6713  just on this one - shouldn't it be that instead ?

```suggestion
                canLoadPosts.current = (result.posts?.length ?? 0 ) > 0;
```

Reply:
If I understood the logic correctly, no. If `result.post` is undefined (not sure when that happens) you can still load posts. Is only when it is defined and length 0.

I would rewrite this logic so it is a bit more clear to the reader.

Reply:
1. If there is an error, you cannot load more posts
2. If the post length is greater than 0, you can load more posts.

I took the code from [here ](https://github.com/mattermost/mattermost-mobile/blob/e443a692656ad10bcb144f9ffe8bf1ab6954ea2a/app/screens/channel/channel_post_list/channel_post_list.tsx#L54-L56)

Should I make the change in both places?

Reply:
We can replace the whole block with 

`canLoadPosts.current = Boolean(result?.post?.length);`